File: frontend/vicourses_mobile_app/lib/models/payment.dart
class Payment {
  final String id;
  final String userId;
  final String username;
  final String email;
  final String courseId;
  final String courseName;
  final String? paypalOrderId;

  Payment({
    required this.id,
    required this.userId,
    required this.username,
    required this.email,
    required this.courseId,
    required this.courseName,
    required this.paypalOrderId,
  });

  static Payment fromMap(Map<String, dynamic> data) {
    return Payment(
      id: data['id'],
      userId: data['userId'],
      username: data['username'],
      email: data['email'],
      courseId: data['courseId'],
      courseName: data['courseName'],
      paypalOrderId: data['paypalOrderId'],
    );
  }
}

File: frontend/vicourses_admin_web/src/types/payment.ts
export type PaymentStatus = "Pending" | "Completed" | "Refunded";
export type PaymentMethod = "Paypal";

export type Payment = {
    id: string;
    userId: string;
    username: string;
    email: string;
    courseId: string;
    courseName: string;
    createdAt: string;
    paymentDueDate: string;
    refundDueDate: string;
    listPrice: number;
    discount: number;
    totalPrice: number;
    couponCode: string | null;
    status: PaymentStatus;
    method: PaymentMethod;
    paypalOrderId: string | null;
}